File: src/Domain/Office/code/Models/Office.cs
  namespace Habitat.Office.Models
{
  using Sitecore.Data.Items;

  public class Office
  {
    public Office(Item item)
    {
      Name = item[Templates.Office.Fields.Name];
      Address = item[Templates.Office.Fields.Address];
      Latitude = item[Templates.Office.Fields.Latitude];
      Longitude = item[Templates.Office.Fields.Longitude];
    }

    public string Name { get; set; }
    public string Address { get; set; }
    public string Latitude { get; set; }
    public string Longitude { get; set; }
  }
}

File: src/Domain/Office/code/Scripts/Office/Office.js
(function (window, $) {
  var map = {};

  window.initMaps = function () {
    initMaps();
    initOffice();
  };

  function initMaps() {
    var elements = $(".map-container");
    $.each(elements, function (index, element) {
      map = new google.maps.Map(element, {
        center: { lat: -5.055576, lng: 3.803790 },
        zoom: 2
      });
    });
  }

  function initOffice() {
    var contextItemId = $("#mapContextItem").val();
    if (!contextItemId) {
      return;
    }

    $.ajax(
    {
      url: "/api/Office/GetOfficeCoordinates",
      method: "POST",
      data: {
        itemId: contextItemId
      },
      success: function (data) {        
        var infoWindow = new google.maps.InfoWindow({
          content: ""
        });
        $.each(data, function (index, office) {
          var contentString = '<h1>' + office.Name + '</h1>' +
            '<p>' + office.Address + '</p>';          
          var marker = new google.maps.Marker({
            position: { lat: parseFloat(office.Latitude), lng: parseFloat(office.Longitude) },
            map: map,
            title: office.Name
          });
          google.maps.event.addListener(marker, 'click', function () {
            infoWindow.setContent(contentString);
            infoWindow.open(map, marker);
          });
        });
      }
    });
  }
})(window, jQuery);